src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListView/ListView.cs

```diff
@@ -4835,7 +4835,7 @@ protected virtual void OnItemChecked(ItemCheckedEventArgs e)
             return;
         }
 
-        if (IsAccessibilityObjectCreated)
+        if (IsAccessibilityObjectCreated && e.Item.ListView == this)
```

Comment:
```suggestion
        if (e.Item.ListView == this && IsAccessibilityObjectCreated)
```

I think this test `e.Item.ListView == this` is faster, so the condition will complete faster if it fails